### ISSUE TITLE
[BUG] `torch_dtype` deprecation warning still emitted on explicit-dtype load and dict-input save

### ISSUE THREAD

**Issue by rrioh** [mlflow/mlflow#25021]
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Issues Policy acknowledgement

- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### MLflow version

- master (reproduced at commit c46e5031f, after #24929 was merged)

### System information

- macOS 15.6.1 (Apple Silicon), Python 3.11, transformers 5.14.1, torch 2.13.0

### Describe the problem

#24929 fixed the `torch_dtype is deprecated` warning reported in #24875 for the common case of loading a saved transformers model, but two paths still emit it with transformers >= 4.56:

1. Loading with an explicit dtype, e.g. `mlflow.transformers.load_model(uri, torch_dtype=...)`. The kwarg is read but not removed from `kwargs`, so `conf.update(**kwargs)` in `_load_model` re-adds the deprecated name and it reaches `transformers.pipeline()`.
2. Saving a components dict that contains a `torch_dtype` entry. `_build_pipeline_from_model_input` forwards the key to `pipeline()` unchanged.

A fix with regression tests is up at #25017.

### Code to reproduce issue

Run each path in a fresh Python process. transformers deduplicates the warning within a process (`logger.warning_once`), so a single script running both paths prints it only once.

Path 1: loading with an explicit dtype

```python
import torch
import transformers
import mlflow

pipe = transformers.pipeline(
    "text-classification", model="distilbert-base-uncased-finetuned-sst-2-english"
)
mlflow.transformers.save_model(transformers_model=pipe, path="model")
mlflow.transformers.load_model("model", torch_dtype=torch.float16)
# -> `torch_dtype` is deprecated! Use `dtype` instead!
```

Path 2: saving a components dict

```python
import torch
import transformers
import mlflow

pipe = transformers.pipeline(
    "text-classification", model="distilbert-base-uncased-finetuned-sst-2-english"
)
mlflow.transformers.save_model(
    transformers_model={
        "model": pipe.model,
        "tokenizer": pipe.tokenizer,
        "torch_dtype": torch.float16,
    },
    task="text-classification",
    path="model2",
)
# -> `torch_dtype` is deprecated! Use `dtype` instead!
```

### Stack trace

No exception. The warning is: `torch_dtype` is deprecated! Use `dtype` instead!

### What component(s) does this bug affect?

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow model serving, deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/uiux`: Front-end, user experience, plotting
- [ ] `area/docs`: MLflow documentation pages
